src/psyclone/psyir/nodes/omp_task_directive.py

```diff
@@ -0,0 +1,2136 @@
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Authors A. B. G. Chalk, STFC Daresbury Lab
+# -----------------------------------------------------------------------------
+""" This module contains the implementation of the OpenMP Task Directive
+node."""
+
+from __future__ import absolute_import
+import itertools
+import math
+
+from psyclone.errors import GenerationError
+from psyclone.psyir.nodes import (
+    Reference,
+    Assignment,
+    IfBlock,
+    ArrayReference,
+    ArrayOfStructuresReference,
+    StructureReference,
+    Call,
+    ArrayMember,
+)
+from psyclone.psyir.nodes.array_mixin import ArrayMixin
+from psyclone.psyir.nodes.array_of_structures_member import (
+    ArrayOfStructuresMember,
+)
+from psyclone.psyir.nodes.operation import BinaryOperation
+from psyclone.psyir.nodes.loop import Loop
+from psyclone.psyir.nodes.literal import Literal
+from psyclone.psyir.nodes.ranges import Range
+from psyclone.psyir.nodes.member import Member
+from psyclone.psyir.nodes.omp_clauses import (
+    OMPPrivateClause,
+    OMPFirstprivateClause,
+    OMPDependClause,
+    OMPSharedClause,
+)
+from psyclone.psyir.nodes.omp_directives import (
+    OMPRegionDirective,
+    OMPSingleDirective,
+    OMPParallelDirective,
+)
+from psyclone.psyir.nodes.schedule import Schedule
+from psyclone.psyir.symbols import INTEGER_TYPE, DataSymbol
+
+
+class OMPTaskDirective(OMPRegionDirective):
+    """
+    Class representing an OpenMP TASK directive in the PSyIR after lowering.
+    This node should not be created by any transformation, and it solely used
+    to represent TASK directives after lowering a DynamicOMPTaskDirective.
+
+    :param list children: list of Nodes that are children of this Node.
```

Comment:
Please give explicit type information for this, i.e.:

    :type children: List[:py:class:`psyclone.psyir.nodes.Node`]

Reply:
Done